Fail to add item

elrayle opened the pull request:
Adds new tests in association_spec:
- it "should return an updated array of generic_files"
- it "should return an updated array of generic_files when generic_files is empty"

NOTE: In both cases the association container is unchanged.  In the first test, there are two generic_files in the container at the start of the test and two generic_files in the container at the end of the test.  For the second test, the container is empty at the start and end.

elrayle committed:
test for BUG fail_to_add

elrayle committed:
fix text comment

jcoyne commented on spec/activefedora/association_spec.rb:
```diff
@@ -37,6 +40,20 @@ class Image < ActiveFedora::Base
+        expect(image.generic_files).to eq [generic_file2, generic_file1, generic_file3]
+      end
+
+      it "should return an updated array of generic_files when generic_files is empty" do
```
Both tests seem to check the same failure. Can we remove one of them?

elrayle replied:
They are checking the same thing from a different starting point.  I am not sure if there is one or multiple bugs.

jcoyne replied:
If we don't know, lets remove one. It seems like the fix made both pass.

terrellt replied:
The same fix took care of both issues, so just keeping one should be fine.

terrellt commented on spec/activefedora/association_spec.rb:
```diff
@@ -37,6 +40,20 @@ class Image < ActiveFedora::Base
+        current_generic_files = image2.generic_files.container.to_a
+        new_generic_files = current_generic_files + [generic_file3]
+        image.generic_files = new_generic_files
+        expect(image2.generic_files).to eq [generic_file3]
```
@elrayle I don't understand what this test is supposed to do. You add a file to one image, and expect it to show up in another?

Commit:
Create new proxies.

This is heavy-handed, and temporary to unblock PCDM.

Commit:
Remove binding.pry

Commit:
Wait for Solr to Start

This is better than some arbitrary wait time.

elrayle committed:
FIX test - consistently use image2 for empty test

elrayle commented:
@terrellt That was a typo in the second test.  It should have set the new_generic_files on image2.  The use of image2 was to be sure generic_files was empty.

terrellt commented:
Oh good, passes now then. @jcoyne Can you review?

jcoyne commented on spec/activefedora/association_spec.rb:
```diff
@@ -25,10 +26,12 @@ class Image < ActiveFedora::Base
     before do
       image.generic_files = [generic_file2, generic_file1]
       image.save
+      image2.save
```
Why are we saving here? It saves two times. Calling `let!(:image2)...` would be preferable.

jcoyne commented:
Can you rebase this against current master? I'm a bit confused about the net changes?

jcoyne commented on lib/active_fedora/aggregation/proxy_container.rb:
```diff
@@ -52,6 +57,7 @@ def link_target(new_proxies)
     # TODO clear out the old proxies (or reuse them)
```
Remove this. It looks like it's no longer TODO

elrayle committed:
removed update array when starting empty

Per Justin, removing test which duplicates "should return an updated array of generic_files", albeit with a different starting state.  This avoids slow tests.

elrayle commented:
The branch got messed up.  It was replaced by [FAIL_to_add_items](https://github.com/projecthydra-labs/activefedora-aggregation/tree/FAIL_to_add_items),
